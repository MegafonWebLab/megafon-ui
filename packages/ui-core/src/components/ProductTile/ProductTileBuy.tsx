import * as React from 'react';
import * as PropTypes from 'prop-types';
import { cnCreate } from '../../utils/cn';
import './style/ProductTileBuy.less';
import Button, { IButtonProps } from '../Button/Button';
import TextLink from '../TextLink/TextLink';
import { TLinkTargetType } from './ProductTile';

export interface IProductTileBuyProps {
    /** Class name */
    className?: string;
    /** Buy link */
    buyLink?: string;
    /** Buy Link target */
    buyLinkTarget?: TLinkTargetType;
    /** Buy button text */
    buyButtonText?: string;
    /** Show buy button */
    showBuyButton?: boolean;
    /** Connect link */
    connectLink?: string;
    /** Connect Link target */
    connectLinkTarget?: TLinkTargetType;
    /** Connect button text */
    connectButtonText?: string;
    /** Show connect button */
    showConnectButton?: boolean;
    /** button border */
    buttonBorder?: IButtonProps['border'];
    /** button font color */
    buttonFontColor?: IButtonProps['fontColor'];
    /** button background color */
    buttonPassiveColor?: IButtonProps['passiveColor'];
    /** Connect hander */
    onClickConnect?(e: React.SyntheticEvent<EventTarget>): void;
    /** Buy hander */
    onClickBuy?(e: React.SyntheticEvent<EventTarget>): void;
}

const cn = cnCreate('mfui-product-tile-buy');
class ProductTileBuy extends React.Component<IProductTileBuyProps> {
    static propTypes = {
        buyLink: PropTypes.string,
        buyLinkTarget: PropTypes.oneOf(['_self', '_blank', '_parent', '_top']),
        buyButtonText: PropTypes.string,
        showBuyButton: PropTypes.bool,
        connectLink: PropTypes.string,
        connectLinkTarget: PropTypes.oneOf(['_self', '_blank', '_parent', '_top']),
        connectButtonText: PropTypes.string,
        showConnectButton: PropTypes.bool,
        buttonBorder: Button.propTypes.border,
        buttonFontColor: Button.propTypes.fontColor,
        buttonPassiveColor: Button.propTypes.passiveColor,
        onClickBuy: PropTypes.func,
        onClickConnect: PropTypes.func,
    };

    static defaultProps: Partial<IProductTileBuyProps> = {
        buyButtonText: 'Купить',
        showBuyButton: true,
        connectButtonText: 'Перейти на тариф',
        showConnectButton: true,
        buyLinkTarget: '_self',
        connectLinkTarget: '_self',
    };

    render() {
        const {
            className,
            buyLink,
            buyButtonText,
            showBuyButton,
            connectLink,
            connectButtonText,
            showConnectButton,
            buttonBorder,
            buttonFontColor,
            buttonPassiveColor,
            onClickBuy,
            onClickConnect,
            buyLinkTarget,
            connectLinkTarget,
        } = this.props;

        return (
            <div className={cn('', {}, className)}>
                {showBuyButton &&
                    <Button
                        className={cn('button', { 'single-button': !showConnectButton })}
                        passiveColor={buttonPassiveColor}
                        hoverColor="green"
                        sizeAll="medium"
                        href={buyLink}
                        target={buyLinkTarget}
                        border={buttonBorder}
                        fontColor={buttonFontColor}
                        onClick={onClickBuy}
                    >
                        {buyButtonText}
                    </Button>
                }
                {showConnectButton &&
                    <TextLink
                        className={cn('detail-link')}
                        href={connectLink}
                        target={connectLinkTarget}
                        onClick={onClickConnect}
                    >
                        {connectButtonText}
                    </TextLink>
                }
            </div>
        );
    }
}

export default ProductTileBuy;
