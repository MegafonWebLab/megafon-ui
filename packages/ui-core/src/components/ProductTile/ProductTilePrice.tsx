import * as React from 'react';
import * as PropTypes from 'prop-types';
import { cnCreate } from '../../utils/cn';
import './style/ProductTilePrice.less';
import AnimationValue from './ProductTileValue';

interface IProductTilePriceProps {
    /** title */
    title?: string;
    /** value */
    value: string;
    /** unitExtra */
    unitExtra: string;
    /** unitValue */
    unitValue: string;
    /** discount */
    discount?: string;
}

const cn = cnCreate('mfui-product-tile-price');
class ProductTilePrice extends React.Component<IProductTilePriceProps> {
    static propTypes = {
        title: PropTypes.string,
        value: PropTypes.string.isRequired,
        unitExtra: PropTypes.string,
        unitValue: PropTypes.string,
        discount: PropTypes.string,
    };

    render() {
        const { title, value, unitExtra, unitValue, discount } = this.props;
        const valueElem = <AnimationValue value={(discount || value) + ` ${unitValue}`} />;
        const unitElem = <span className={cn('payment-period')}>{unitExtra}</span>;

        return (
            <div className={cn('', { discount: !!discount })}>
                <span className={cn('old-price')}>
                    <span className={cn('old-price-value')}>
                        {discount ? `${value} ${unitValue}` : ' '}
                    </span>
                </span>
                <span className={cn('actual-price')}>
                    {valueElem} {unitElem}
                </span>
                <div className={cn('discount-condition')}>{title}</div>
            </div>
        );
    }
}

export default ProductTilePrice;
