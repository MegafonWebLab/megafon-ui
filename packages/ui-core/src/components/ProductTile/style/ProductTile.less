@import '~styles/base.less';

@b: e('.mfui-product-tile');

@{b} {
    display: flex;
    flex: 1 1 100%;
    flex-direction: column;
    justify-content: space-between;
    box-sizing: border-box;
    position: relative;
    border-radius: 10px;
    box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.2);
    background-color: @clearWhite;
    padding: 32px 8px 16px;

    &:hover {
        box-shadow: 0 4px 10px 0 rgba(0, 0, 0, 0.2);
    }

    @media @mobileSM {
        &:not(:last-child) {
            margin-bottom: 24px;
        }
    }

    @media @mobileB {
        padding-top: 46px;
    }

    @media @desktopS {
        padding-top: 34px;
    }

    @media @desktopMU {
        padding: 34px 12px 20px;
    }

    &__content {
        margin-bottom: 25px;
    }

    &__constructor {
        background-color: lighten(@spbSky1, 0.5);
        padding: 24px 16px 16px;
        margin-top: 10px;
        position: relative;

        &:before {
            border-top-color: lighten(@spbSky1, 0.5);
            border-left-color: lighten(@spbSky1, 0.5);
            border-bottom-color: transparent;
            border-right-color: transparent;
            border-width: 5px;
            border-style: solid;
            content: '';
            left: 50%;
            position: absolute;
            top: -4px;
            transform: translateX(-50%) rotate(45deg);
        }

        @media @mobileB {
            padding: 16px 8px;
        }

        @media @desktopS {
            padding: 22px 8px;
        }

        @media @desktopMU {
            padding-left: 12px;
            padding-right: 12px;
        }
    }

    &__title {
        font-size: 20px;
        font-weight: 500;
        text-align: center;
        line-height: 28px;
        letter-spacing: 0.5px;
        min-height: 47px;

        @media @mobileSM {
            min-height: auto;
        }

        @media @mobileB {
            font-size: 18px;
            line-height: 24px;
        }

        @media @desktopS {
            font-size: 18px;
            line-height: 24px;
        }

        @media @desktopMU {
            min-height: 56px;
        }
    }

    &__link-wrapper {
        text-align: center;
        margin-bottom: 18px;

        @media @mobileSM {
            margin-bottom: 17px;
        }

        @media @desktopS {
            margin-bottom: 26px;
        }
    }

    &_active {
        box-shadow: inset 0 0 0 2px @greenActive, 0 4px 10px 0 rgba(0, 0, 0, 0.2);
    }

    &_active:hover {
        box-shadow: inset 0 0 0 2px @greenActive, 0 4px 10px 0 rgba(0, 0, 0, 0.2);
    }
}
