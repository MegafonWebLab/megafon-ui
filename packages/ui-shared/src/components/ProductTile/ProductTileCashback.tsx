import * as React from 'react';
import * as PropTypes from 'prop-types';
import { cnCreate } from '@megafon/ui-core';
import './style/ProductTileCashback.less';

interface IProductTileCashbackProps {
    /** title */
    title?: string;
    /** value */
    value?: number;
    /** unit */
    unit?: string;
}

const cn = cnCreate('mfui-product-tile-cashback');
class ProductTileCashback extends React.Component<IProductTileCashbackProps> {
    static propTypes = {
        title: PropTypes.string,
        value: PropTypes.number,
        unit: PropTypes.string,
    };

    render() {
        const { title, value, unit } = this.props;

        if (!title && !value && !unit) {
            return null;
        }

        return (
            <div className={cn('')}>
                <span className={cn('text')}>
                    {title} {value} {unit}
                </span>
            </div>
        );
    }
}

export default ProductTileCashback;
