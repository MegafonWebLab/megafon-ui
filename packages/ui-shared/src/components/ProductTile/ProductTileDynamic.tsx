import * as React from 'react';
import * as PropTypes from 'prop-types';
import { cnCreate } from '@megafon/ui-core';
import './style/ProductTileDynamic.less';
import ProductSwitcher from '../ProductSwitcher/ProductSwitcher';
import { IServicePack, ISwitcher } from './ProductTile';
import AnimationValue from './ProductTileValue';

interface IProductTileDynamicProps {
    /** Current pack */
    currentPack: Partial<IServicePack>;
    /** Switcher */
    switcher: ISwitcher;
    /** Start calls index */
    startCallsIndex?: number;
    /** Start traffic index */
    startTrafficIndex?: number;
    /** Change calls */
    onChangeCalls(e: React.SyntheticEvent<EventTarget>, value: string, index: number): boolean;
    /** Change Traffic */
    onChangeTraffic(e: React.SyntheticEvent<EventTarget>, value: string, index: number): boolean;
}

const cn = cnCreate('mfui-product-tile-dynamic');
class ProductTileDynamic extends React.Component<IProductTileDynamicProps> {
    static propTypes = {
        currentPack: PropTypes.shape({
            calls: PropTypes.shape({
                value: PropTypes.number,
                unit: PropTypes.string,
            }),
            traffic: PropTypes.shape({
                value: PropTypes.number,
                unit: PropTypes.string,
            }),
        }),
        switcher: PropTypes.shape({
            calls: PropTypes.arrayOf(PropTypes.string),
            traffic: PropTypes.arrayOf(PropTypes.string),
        }),
        startCallsIndex: PropTypes.number,
        startTrafficIndex: PropTypes.number,
        onChangeCalls: PropTypes.func,
        onChangeTraffic: PropTypes.func,
    };

    render() {
        const {
            currentPack,
            switcher,
            startCallsIndex,
            startTrafficIndex,
            onChangeTraffic,
            onChangeCalls,
        } = this.props;

        return (
            <div className={cn('')}>
                <div className={cn('constructor-pack')}>
                    <AnimationValue
                        hAlign="center"
                        value={`${(currentPack.calls || {}).value} ${(currentPack.calls || {}).unit}`}
                    />
                </div>
                <ProductSwitcher
                    className={cn('constructor-range')}
                    startIndex={startCallsIndex}
                    theme="tariff-showcase"
                    items={switcher.calls.map((value: string) => ({ title: value, value, unit: value }))}
                    onChange={onChangeCalls}
                />
                <ProductSwitcher
                    className={cn('constructor-range')}
                    startIndex={startTrafficIndex}
                    theme="tariff-showcase"
                    items={switcher.traffic.map((value: string) => ({ title: value, value, unit: value }))}
                    onChange={onChangeTraffic}
                />
                <div className={cn('constructor-pack')}>
                    <AnimationValue
                        hAlign="center"
                        value={`${(currentPack.traffic || {}).value} ${(currentPack.traffic || {}).unit}`}
                    />
                </div>
            </div>
        );
    }
}

export default ProductTileDynamic;
