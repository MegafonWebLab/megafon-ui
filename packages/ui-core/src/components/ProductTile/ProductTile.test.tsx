import * as React from 'react';
import { shallow } from 'enzyme';
import ProductTile, { IProductTileProps, IServicePack } from './ProductTile';
import MegafonTv from 'icons/Service-logos/24/MF-TV.svg';

const tariff: Partial<IProductTileProps> = {
    buyButtonText: 'sdfsdf',
    connectButtonText: 'sdfsdf',
    secondParamsHead: 'sdfsdf',
    showConnectButton: true,
    title: 'Включайся! Смотри',
    topBadgeTitle: 'sdfsdf',
    link: '/tariffs/vklyuchaysya/smotri.html',
    buyLink: '/zakaz/?tariff=look',
    payment: {
        value: '1000',
        unitExtra: 'за 30 дней',
        unitValue: '₽',
        discount: '0',
    },
    packs: [
        {
            value: 1500,
            unit: 'минут',
            title: 'title',
            isDelim: true,
        },
        {
            value: 100,
            unit: 'смс',
            title: 'title',
            isDelim: true,
        },
        {
            value: 3,
            unit: 'ГБ',
            title: 'title',
            isDelim: false,
        },
    ],
    firstParams: {
        items: [
            {
                title: 'VIP-тариф',
                caption: 'Привилегии обслуживания и другое',
                svgIcon: <MegafonTv />,
            },
            {
                title: 'МегаФон ТВ',
                caption:
                    'Безлимитный трафик и пакеты кино, сериалов и ТВ-каналов на любой вкус',
                svgIcon: <MegafonTv />,
            },
        ],
    },
    secondParams: [
        {
            title: 'Кэшбэк 20%',
            footnote: 'Участвуйте в программе и тратьте кэшбэк',
            value: '0',
            unit: '',
            svgIcon: <MegafonTv />,
        },
        {
            title: 'Безлимитные SMS',
            footnote: '',
            value: '55',
            unit: '₽ за 30 дней',
            svgIcon: <MegafonTv />,
        },
    ],
    cashback: {
        title: 'Кэшбэк',
        value: 20,
        unit: '%',
    },
    info: {
        billingIds: [],
        sms: '05007897',
        smsText: 'ДА',
        smsSite: null,
        ivr: '05007897',
        ussd: '<span class="c-number-call">sdfsdf</span>',
        lkLink: 'https://lk.megafon.ru/tariffs/',
        title: 'Включайся! Смотри',
        id: 1598,
    },
};

const servicePacks: Array<Partial<IServicePack>> = [
    {
        buyLink: '&options=null',
        shopTag: 'tag1',
        calls: {
            value: 300,
            unit: 'минут',
        },
        traffic: {
            value: 6,
            unit: 'ГБ',
        },
        payment: {
            value: '400',
            discount: '200',
        },
        options: [
            {
                title: 'Безлимитные SMS',
                value: '55',
                unit: '₽ за 30 дней',
                footnote: '',
                svgIcon: <MegafonTv />,
            },
            {
                title: 'Кэшбэк 20%',
                value: '0',
                unit: '',
                footnote:
                    'Участвуйте в программе и трпки в салонах и интернет-магазине МегаФона',
                svgIcon: <MegafonTv />,
            },
        ],
    },
    {
        'buyLink': '&options=null',
        'calls': {
            'value': 600,
            'unit': 'минут',
        },
        'traffic': {
            'value': 18,
            'unit': 'ГБ',
        },
        'payment': {
            'value': '600',
            'discount': '0',
        },
        'options': [
            {
                'title': 'Безлимитные SMS',
                'value': '55',
                'unit': '₽ за 30 дней',
                'footnote': '',
                'svgIcon': <MegafonTv />,
            },
            {
                'title': 'Кэшбэк 20%',
                'value': '0',
                'unit': '',
                'footnote': 'Участвуйте в программе и тратьте кэшбэк ',
                'svgIcon': <MegafonTv />,
            },
        ],
    },
    {
        buyLink: '&options=null',
        shopTag: 'tag2',
        calls: {
            value: 650,
            unit: 'минут',
        },
        traffic: {
            value: 6,
            unit: 'ГБ',
        },
        payment: {
            value: '590',
            discount: '300',
        },
        options: [
            {
                title: 'Безлимитные SMS',
                value: '55',
                unit: '₽ за 30 дней',
                footnote: '',
                svgIcon: <MegafonTv />,
            },
            {
                title: 'Кэшбэк 20%',
                value: '0',
                unit: '',
                footnote:
                    'Участвуйте тратьте кэшбэк на сервисы и покупки в салонах и магазине МегаФона',
                svgIcon: <MegafonTv />,
            },
        ],
    },
    {
        buyLink: '&options=null',
        shopTag: 'tag3',
        calls: {
            value: 300,
            unit: 'минут',
        },
        traffic: {
            value: 18,
            unit: 'ГБ',
        },
        payment: {
            value: '590',
            discount: '',
        },
        options: [
            {
                title: 'Безлимитные SMS',
                value: '55',
                unit: '₽ за 30 дней',
                footnote: '',
                svgIcon: <MegafonTv />,
            },
            {
                title: 'Кэшбэк 20%',
                value: '0',
                unit: '',
                footnote:
                    'Участвуйте в купки в салонах и интернет-магазине МегаФона',
                svgIcon: <MegafonTv />,
            },
        ],
    },
    {
        buyLink: '&options=null',
        calls: {
            value: 750,
            unit: 'минут',
        },
        traffic: {
            value: 6,
            unit: 'ГБ',
        },
        payment: {
            value: '670',
            discount: '',
        },
        options: [
            {
                title: 'Безлимитные SMS',
                value: '55',
                unit: '₽ за 30 дней',
                footnote: '',
                svgIcon: <MegafonTv />,
            },
            {
                title: 'Кэшбэк 20%',
                value: '0',
                unit: '',
                footnote: 'Укупки в салонах и интернет-магазине МегаФона',
                svgIcon: <MegafonTv />,
            },
        ],
    },
    {
        buyLink: '&options=null',
        calls: {
            value: 900,
            unit: 'минут',
        },
        traffic: {
            value: 28,
            unit: 'ГБ',
        },
        payment: {
            value: '770',
            discount: '',
        },
        options: [
            {
                title: 'Безлимитные SMS',
                value: '55',
                unit: '₽ за 30 дней',
                footnote: '',
                svgIcon: <MegafonTv />,
            },
            {
                title: 'Кэшбэк 20%',
                value: '0',
                unit: '',
                footnote: 'Участвуки в салонах и интернет-магазине МегаФона',
                svgIcon: <MegafonTv />,
            },
        ],
    },
];

const target = {
    target: {},
} as React.SyntheticEvent<EventTarget>;

describe('<ProductTile />', () => {
    describe('snapshots', () => {
        it('should render ProductTile', () => {
            const wrapper = shallow(
                <ProductTile
                    {...tariff}
                    servicePacks={servicePacks}
                    startCallsIndex={2}
                    startTrafficIndex={1}
                />
            );
            expect(wrapper).toMatchSnapshot();
        });

        it('should render ProductTile with cookies', () => {
            const wrapper = shallow(
                <ProductTile
                    {...tariff}
                    servicePacks={servicePacks}
                    startCallsIndex={2}
                    startTrafficIndex={1}
                    cookieCallsIndex={4}
                    cookieTrafficIndex={2}
                />
            );
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('onCallsChange tests', () => {
        it('should call onCallsChange when handleChangeCalls fired and set state.callsIndex', () => {
            const positionIndex = 2;
            const [ pack ] = servicePacks;
            const onCallsChange = jest.fn();
            const wrapper = shallow<ProductTile>(
                <ProductTile
                    {...tariff}
                    servicePacks={servicePacks}
                    startCallsIndex={2}
                    startTrafficIndex={1}
                    onCallsChange={onCallsChange}
                />
            );

            wrapper.instance().handleChangeCalls(target, pack.calls!.value.toString(), positionIndex);
            expect(onCallsChange).toBeCalledWith(positionIndex);
            expect(wrapper.state('callsIndex')).toBe(positionIndex);
        });

        it('shouldn\'t call onCallsChange when handleChangeCalls return false', () => {
            const [ , pack ] = servicePacks;
            const onTrafficChange = jest.fn();
            const wrapper = shallow<ProductTile>(
                <ProductTile
                    {...tariff}
                    servicePacks={servicePacks}
                    startCallsIndex={2}
                    startTrafficIndex={1}
                    onTrafficChange={onTrafficChange}
                />
            );

            wrapper.instance().handleChangeTraffic(target, pack.calls!.value.toString(), 2);
            expect(onTrafficChange).not.toBeCalled();
        });
    });

    describe('onTrafficChange tests', () => {
        it('should call onTrafficChange when handleChangeTraffic fired and set state.trafficIndex', () => {
            const positionIndex = 2;
            const [ pack ] = servicePacks;
            const onTrafficChange = jest.fn();
            const wrapper = shallow<ProductTile>(
                <ProductTile
                    {...tariff}
                    servicePacks={servicePacks}
                    startCallsIndex={2}
                    startTrafficIndex={1}
                    onTrafficChange={onTrafficChange}
                />
            );

            wrapper.instance().handleChangeTraffic(target, pack.traffic!.value.toString(), positionIndex);
            expect(onTrafficChange).toBeCalledWith(positionIndex);
            expect(wrapper.state('trafficIndex')).toBe(positionIndex);
        });

        it('shouldn\'t call onTrafficChange when handleChangeTraffic return false', () => {
            const [ , pack ] = servicePacks;
            const onTrafficChange = jest.fn();
            const wrapper = shallow<ProductTile>(
                <ProductTile
                    {...tariff}
                    servicePacks={servicePacks}
                    startCallsIndex={2}
                    startTrafficIndex={1}
                    onTrafficChange={onTrafficChange}
                />
            );

            wrapper.instance().handleChangeTraffic(target, pack.traffic!.value.toString(), 2);
            expect(onTrafficChange).not.toBeCalled();
        });
    });
});
