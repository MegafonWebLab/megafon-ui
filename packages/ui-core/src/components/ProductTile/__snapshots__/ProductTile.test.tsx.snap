// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTile /> snapshots should render ProductTile 1`] = `
<div
  className="mfui-product-tile mfui-product-tile_constructor"
>
  <ProductTileHint
    linkTarget="_self"
    title="sdfsdf"
  />
  <div
    className="mfui-product-tile__content"
  >
    <h2
      className="mfui-product-tile__title"
    >
      <TextLink
        className="mfui-product-tile__title-link"
        color="black"
        href="/tariffs/vklyuchaysya/smotri.html"
        onClick={[Function]}
        target="_self"
        underlineStyle="none"
        underlineVisibility="hover"
      >
        Включайся! Смотри
      </TextLink>
    </h2>
    <div
      className="mfui-product-tile__link-wrapper"
    >
      <TextLink
        className="mfui-product-tile__detail-link"
        color="blue"
        href="/tariffs/vklyuchaysya/smotri.html"
        onClick={[Function]}
        target="_self"
        underlineStyle="solid"
        underlineVisibility="hover"
      >
        Подробнее
      </TextLink>
    </div>
    <ProductTileCashback
      title="Кэшбэк"
      unit="%"
      value={20}
    />
    <ProductTilePrice
      discount="0"
      unitExtra="за 30 дней"
      unitValue="₽"
      value="1000"
    />
    <div
      className="mfui-product-tile__constructor"
    >
      <ProductTileDynamic
        currentPack={Object {}}
        onChangeCalls={[Function]}
        onChangeTraffic={[Function]}
        startCallsIndex={2}
        startTrafficIndex={1}
        switcher={
          Object {
            "calls": Array [
              "300",
              "600",
              "650",
              "750",
              "900",
            ],
            "traffic": Array [
              "6",
              "18",
              "28",
            ],
          }
        }
      />
      <ProductTileOptions
        options={
          Array [
            Object {
              "caption": "Привилегии обслуживания и другое",
              "svgIcon": <test-file-stub />,
              "title": "VIP-тариф",
            },
            Object {
              "caption": "Безлимитный трафик и пакеты кино, сериалов и ТВ-каналов на любой вкус",
              "svgIcon": <test-file-stub />,
              "title": "МегаФон ТВ",
            },
          ]
        }
      />
    </div>
    <ProductTileOptions
      head="sdfsdf"
      onClickBubble={[Function]}
      options={
        Array [
          Object {
            "footnote": "Участвуйте в программе и тратьте кэшбэк",
            "svgIcon": <test-file-stub />,
            "title": "Кэшбэк 20%",
            "unit": "",
            "value": "0",
          },
          Object {
            "footnote": "",
            "svgIcon": <test-file-stub />,
            "title": "Безлимитные SMS",
            "unit": "₽ за 30 дней",
            "value": "55",
          },
        ]
      }
    />
  </div>
  <ProductTileBuy
    buyButtonText="sdfsdf"
    buyLink="/zakaz/?tariff=look"
    buyLinkTarget="_self"
    connectButtonText="sdfsdf"
    connectLinkTarget="_self"
    onClickBuy={[Function]}
    onClickConnect={[Function]}
    showBuyButton={true}
    showConnectButton={true}
  />
</div>
`;

exports[`<ProductTile /> snapshots should render ProductTile with cookies 1`] = `
<div
  className="mfui-product-tile mfui-product-tile_constructor"
>
  <ProductTileHint
    linkTarget="_self"
    title="sdfsdf"
  />
  <div
    className="mfui-product-tile__content"
  >
    <h2
      className="mfui-product-tile__title"
    >
      <TextLink
        className="mfui-product-tile__title-link"
        color="black"
        href="/tariffs/vklyuchaysya/smotri.html"
        onClick={[Function]}
        target="_self"
        underlineStyle="none"
        underlineVisibility="hover"
      >
        Включайся! Смотри
      </TextLink>
    </h2>
    <div
      className="mfui-product-tile__link-wrapper"
    >
      <TextLink
        className="mfui-product-tile__detail-link"
        color="blue"
        href="/tariffs/vklyuchaysya/smotri.html"
        onClick={[Function]}
        target="_self"
        underlineStyle="solid"
        underlineVisibility="hover"
      >
        Подробнее
      </TextLink>
    </div>
    <ProductTileCashback
      title="Кэшбэк"
      unit="%"
      value={20}
    />
    <ProductTilePrice
      discount="0"
      unitExtra="за 30 дней"
      unitValue="₽"
      value="770"
    />
    <div
      className="mfui-product-tile__constructor"
    >
      <ProductTileDynamic
        currentPack={
          Object {
            "buyLink": "&options=null",
            "calls": Object {
              "unit": "минут",
              "value": 900,
            },
            "options": Array [
              Object {
                "footnote": "",
                "svgIcon": <test-file-stub />,
                "title": "Безлимитные SMS",
                "unit": "₽ за 30 дней",
                "value": "55",
              },
              Object {
                "footnote": "Участвуки в салонах и интернет-магазине МегаФона",
                "svgIcon": <test-file-stub />,
                "title": "Кэшбэк 20%",
                "unit": "",
                "value": "0",
              },
            ],
            "payment": Object {
              "discount": "",
              "value": "770",
            },
            "traffic": Object {
              "unit": "ГБ",
              "value": 28,
            },
          }
        }
        onChangeCalls={[Function]}
        onChangeTraffic={[Function]}
        startCallsIndex={4}
        startTrafficIndex={2}
        switcher={
          Object {
            "calls": Array [
              "300",
              "600",
              "650",
              "750",
              "900",
            ],
            "traffic": Array [
              "6",
              "18",
              "28",
            ],
          }
        }
      />
      <ProductTileOptions
        options={
          Array [
            Object {
              "caption": "Привилегии обслуживания и другое",
              "svgIcon": <test-file-stub />,
              "title": "VIP-тариф",
            },
            Object {
              "caption": "Безлимитный трафик и пакеты кино, сериалов и ТВ-каналов на любой вкус",
              "svgIcon": <test-file-stub />,
              "title": "МегаФон ТВ",
            },
          ]
        }
      />
    </div>
    <ProductTileOptions
      head="sdfsdf"
      onClickBubble={[Function]}
      options={
        Array [
          Object {
            "footnote": "",
            "svgIcon": <test-file-stub />,
            "title": "Безлимитные SMS",
            "unit": "₽ за 30 дней",
            "value": "55",
          },
          Object {
            "footnote": "Участвуки в салонах и интернет-магазине МегаФона",
            "svgIcon": <test-file-stub />,
            "title": "Кэшбэк 20%",
            "unit": "",
            "value": "0",
          },
        ]
      }
    />
  </div>
  <ProductTileBuy
    buyButtonText="sdfsdf"
    buyLink="&options=null"
    buyLinkTarget="_self"
    connectButtonText="sdfsdf"
    connectLinkTarget="_self"
    onClickBuy={[Function]}
    onClickConnect={[Function]}
    showBuyButton={true}
    showConnectButton={true}
  />
</div>
`;
